src/solve.jl

```diff
@@ -178,7 +178,7 @@ function DiffEqBase.__init(prob::Union{DiffEqBase.AbstractODEProblem,DiffEqBase.
   # dtmin is all abs => does not care about sign already.
 
   if !isdae && isinplace(prob) && typeof(u) <: AbstractArray && eltype(u) <: Number && uBottomEltypeNoUnits == uBottomEltype && tType == tTypeNoUnits # Could this be more efficient for other arrays?
-    if !(typeof(u) <: ArrayPartition)
+    if typeof(u) <: ArrayPartition
```

Comment:
This doesn't seem like a good idea. Before, everything apart from ArrayPartition ended up in this branch, and with this change only ArrayPartition would end up in this branch. It seems what you would like is rather to remove the other branch and always use `rate_prototype = recursivecopy(u)`.